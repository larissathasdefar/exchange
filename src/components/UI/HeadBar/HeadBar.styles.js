import styled from 'styled-components'
import Typography from '@material-ui/core/Typography'

export const Container = styled.div`
  min-height: 50px;
  padding: 8px 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #FFFFFF;
  margin-bottom: 12px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 2px 1px -1px rgba(0,0,0,0.12);
`
export const Menu = styled.div`
  display: flex;
  align-items: center;
  min-width: 200px;
  justify-content: space-around;
`

export const CurrentPage = styled(Typography)`
  border-bottom: 2px solid #e64a19;
  cursor: pointer;
`

export const Avatar = styled.div`
  height: 38px;
  width: 38px;
  background-image: url(${({ image }) => image});
  background-size: cover;
  border-radius: 50%;
`

export const Logo = styled.img`
  cursor: pointer;
  -webkit-user-drag: none;
`

export const Link = styled(Typography)`
  cursor: pointer;
`
