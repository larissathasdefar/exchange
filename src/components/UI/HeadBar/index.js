import React, { PureComponent } from 'react'
import _ from 'prop-types'
import LogoImage from 'assets/revolut.png'
import AccountIcon from '@material-ui/icons/AccountCircle'
import Tooltip from '@material-ui/core/Tooltip'
import {
  Container,
  Menu,
  CurrentPage,
  Avatar,
  Logo,
  Link,
} from './HeadBar.styles'

class HeadBar extends PureComponent {
  state = {
    open: false,
  }

  render() {
    const { user, history } = this.props
    const redirect = () => history.push('/pockets')
    const image = user.photo !== ''
      ? <Avatar image={user.photo} alt={user.name} />
      : <AccountIcon fontSize="large" />
    return (
      <Container>
        <Logo src={LogoImage} onClick={redirect} />
        <Menu>
          <Link onClick={redirect}>Bank</Link>
          <CurrentPage onClick={redirect}>
            Exchange
          </CurrentPage>
          {
            user.name !== ''
              ? (
                <Tooltip title={user.name}>
                  {image}
                </Tooltip>
              )
              : image
          }
        </Menu>
      </Container>
    )
  }
}

HeadBar.propTypes = {
  user: _.shape({
    name: _.string,
    photo: _.string,
  }),
  history: _.shape({
    push: _.func,
  }).isRequired,
}

export default HeadBar
