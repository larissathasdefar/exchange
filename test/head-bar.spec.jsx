import React from 'react'
import { shallow } from 'enzyme'

import HeadBar from 'ui/HeadBar'

describe('HeadBar', () => {
  it('Empty user', () => {
    const component = shallow(
      <HeadBar
        user={{ name: '', photo: '' }}
        history={{ push: route => console.log(route) }}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })

  it('User without photo', () => {
    const component = shallow(
      <HeadBar
        user={{ name: 'Larissa Farias', photo: '' }}
        history={{ push: route => console.log(route) }}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })

  it('User with photo', () => {
    const component = shallow(
      <HeadBar
        user={{ name: 'Larissa Farias', photo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/640px-React-icon.svg.png' }}
        history={{ push: route => console.log(route) }}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })
})