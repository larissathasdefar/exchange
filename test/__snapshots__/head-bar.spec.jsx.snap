// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadBar Empty user 1`] = `
"<styled.div>
  <styled.img src={{...}} onClick={[Function: redirect]} />
  <styled.div>
    <WithStyles(Typography)>
      Bank
    </WithStyles(Typography)>
    <Styled(WithStyles(Typography)) onClick={[Function: redirect]}>
      Exchange
    </Styled(WithStyles(Typography))>
    <pure(AccountCircleIcon) fontSize=\\"large\\" />
  </styled.div>
</styled.div>"
`;

exports[`HeadBar User with photo 1`] = `
"<styled.div>
  <styled.img src={{...}} onClick={[Function: redirect]} />
  <styled.div>
    <WithStyles(Typography)>
      Bank
    </WithStyles(Typography)>
    <Styled(WithStyles(Typography)) onClick={[Function: redirect]}>
      Exchange
    </Styled(WithStyles(Typography))>
    <WithStyles(Tooltip) title=\\"Larissa Farias\\">
      <styled.div image=\\"https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/640px-React-icon.svg.png\\" alt=\\"Larissa Farias\\" />
    </WithStyles(Tooltip)>
  </styled.div>
</styled.div>"
`;

exports[`HeadBar User without photo 1`] = `
"<styled.div>
  <styled.img src={{...}} onClick={[Function: redirect]} />
  <styled.div>
    <WithStyles(Typography)>
      Bank
    </WithStyles(Typography)>
    <Styled(WithStyles(Typography)) onClick={[Function: redirect]}>
      Exchange
    </Styled(WithStyles(Typography))>
    <WithStyles(Tooltip) title=\\"Larissa Farias\\">
      <pure(AccountCircleIcon) fontSize=\\"large\\" />
    </WithStyles(Tooltip)>
  </styled.div>
</styled.div>"
`;
